my-app/src/components/register/register.js

```diff
@@ -101,12 +143,13 @@ render(){
                 onChange={e => this.setState({user: {...user, type: e.target.value }})}></input>
                 <span class="checkmark"></span>
               </label>
+              </form>
               <button onClick={this.addUser}>Register</button>
               <p className="message">Already registered? <Link to='/login'>Sign In</Link></p>
           </div>
       </div>
-  );
-}
+    );
+  }
 }
 
-export default Register
+export default withRouter(Register);
```

Comment:
I was also looking into using WithRouter, good to see in what situation you are using it in.